datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -320,7 +320,11 @@ func (f *ChainDataFetcher) makeChainEvent(blockNumber uint64) (blockchain.ChainE
 		}
 		for _, r := range results {
 			if r.Result != nil {
-				internalTraces = append(internalTraces, r.Result.(*vm.InternalTxTrace))
+				switch r.Result.(type) {
+				case []vm.CallFrame:
+					cf := r.Result.([]vm.CallFrame)
+					internalTraces = append(internalTraces, cf[0].ToInternalTxTrace())
```

Comment:
Is `r.Result` type a list of CallFrame, not recursive CallFrames?

Reply:
Why only very first element (`cf[0]`) is changed to `InternalTxTrace`? How about others?

Reply:
The call trace is always a tree structure rooted at the single top frame. Refer to geth https://github.com/ethereum/go-ethereum/blob/master/eth/tracers/native/call.go and kaia https://github.com/kaiachain/kaia/blob/83efc71ee33025b64636bf94e81508c59c8cb6ca/blockchain/vm/call_tracer.go implementations

Reply:
@blukat29 Is https://github.com/kaiachain/kaia/pull/16#discussion_r1663477870 applied?

Reply:
Oh, I got it. But, then why is it in the list? Anyway, I got this.

Reply:
@hyeonLewis Yes in the  https://github.com/kaiachain/kaia/pull/16/commits/c2f323e461a61ff2e453cc6e7fe40cf0bce3e172.

Reply:
@blukat29 Am I missing something? I think it hasn't been applied...

Reply:
Thanks!

Reply:
@jeongkyun-oh Oh the callstack is a linear 'stack' where `callstack[0]` is the top frame, `callstack[1]` is the child frame, ... As the EVM executes the callstack gets longer and shorter, while leaving a wide arrays in `callstack[i].Calls`. It could be confusing.